webapp/tests/karma/unit/services/contact-view-model-generator.js

```diff
@@ -196,59 +193,47 @@ describe('ContactViewModelGenerator service', () => {
+        assert.equal(model.children[0].contacts[0].isPrimaryContact, true);
       });
     });
 
     it('child places are sorted in alphabetical order', () => {
       return runPlaceTest([childPlace2, childPlace]).then(model => {
-        assert.equal(model.children.places[0].doc._id, childPlace._id);
-        assert.equal(model.children.places[1].doc._id, childPlace2._id);
-      });
-    });
-
-    it('child persons are sorted in alphabetical order', () => {
-      doc.type = 'star';
-      return runPlaceTest([childPerson2, childPerson]).then(model => {
-        // Remove the primary contact
-        model.children.persons.splice(0, 1);
-        assert.equal(model.children.persons[0].doc._id, childPerson._id);
-        assert.equal(model.children.persons[1].doc._id, childPerson2._id);
+        assert.equal(model.children[1].contacts[0].doc._id, childPlace._id);
+        assert.equal(model.children[1].contacts[1].doc._id, childPlace2._id);
       });
     });
 
     it('when selected doc is a clinic, child places are sorted in alphabetical order (like for other places)', () => {
       doc.type = 'clinic';
       return runPlaceTest([childPlace2, childPlace]).then(model => {
-        assert.equal(model.children.places[0].doc._id, childPlace._id);
-        assert.equal(model.children.places[1].doc._id, childPlace2._id);
+        assert.equal(model.children[1].contacts[0].doc._id, childPlace._id);
+        assert.equal(model.children[1].contacts[1].doc._id, childPlace2._id);
       });
     });
 
-    it('when selected doc is a clinic, child persons are sorted by age', () => {
+    it('when selected contact type specifies sort_by_dob, child persons are sorted by age', () => {
       doc.type = 'clinic';
       return runPlaceTest([childPerson2, childPerson]).then(model => {
-        // Remove the primary contact
-        model.children.persons.splice(0, 1);
-        assert.equal(model.children.persons[0].doc._id, childPerson2._id);
-        assert.equal(model.children.persons[1].doc._id, childPerson._id);
+        assert.equal(model.children[0].contacts.length, 3);
+        assert.equal(model.children[0].contacts[1].doc._id, childPerson2._id);
+        assert.equal(model.children[0].contacts[2].doc._id, childPerson._id);
       });
     });
 
     it('child places and persons get displayed separately', () => {
       return runPlaceTest([childContactPerson, deceasedChildPerson]).then(model => {
-        assert.equal(model.children.persons.length, 1);
-        assert.deepEqual(model.children.persons[0].doc, childContactPerson);
-        assert.equal(model.children.deceased.length, 1);
-        assert.deepEqual(model.children.deceased[0].doc, deceasedChildPerson);
+        assert.equal(model.children[0].contacts.length, 2);
+        assert.deepEqual(model.children[0].contacts[0].doc, childContactPerson);
+        assert.deepEqual(model.children[0].contacts[1].doc, deceasedChildPerson);
```

Comment:
I think this deceivingly named test was testing that deceased persons get displayed separately.
Assertions specifically show that deceased persons are NOT displayed separately.

Reply:
You're right. I'll change it to test things like deceased count.